src/Tokens/TokensRequestor.php

```diff
@@ -106,6 +107,11 @@ private static function tokensFromJsonResponse(stdClass $jsonResponse): Tokens
 
         $accessToken = new AccessToken($jsonResponse->access_token);
 
+        $refreshToken = null;
+        if (isset($jsonResponse->refresh_token)) {
```

Comment:
I think we should also check if it's empty before trying to instantiate a refresh token. Something like `&& trim($jsonResponse->refresh_token) !== '')`. Can you also add a test case for the situation of an empty `refresh_token` string?

Reply:
Good point, I'll do that!